matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/contentscanning/ContentScannerModule.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 New Vector Ltd - All Rights Reserved
+ * Unauthorized copying of this file, via any medium is strictly prohibited
+ * Proprietary and confidential
+ */
+
+package org.matrix.android.sdk.internal.session.contentscanning
+
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import io.realm.RealmConfiguration
+import org.matrix.android.sdk.api.session.contentscanning.ContentScannerService
+import org.matrix.android.sdk.internal.database.RealmKeysUtils
+import org.matrix.android.sdk.internal.di.ContentScannerDatabase
+import org.matrix.android.sdk.internal.di.SessionFilesDirectory
+import org.matrix.android.sdk.internal.di.UserMd5
+import org.matrix.android.sdk.internal.session.SessionModule
+import org.matrix.android.sdk.internal.session.SessionScope
+import org.matrix.android.sdk.internal.session.contentscanning.data.ContentScanningStore
+import org.matrix.android.sdk.internal.session.contentscanning.db.ContentScannerRealmModule
+import org.matrix.android.sdk.internal.session.contentscanning.db.RealmContentScannerStore
+import org.matrix.android.sdk.internal.session.contentscanning.tasks.DefaultDownloadEncryptedTask
+import org.matrix.android.sdk.internal.session.contentscanning.tasks.DefaultGetServerPublicKeyTask
+import org.matrix.android.sdk.internal.session.contentscanning.tasks.DefaultScanEncryptedTask
+import org.matrix.android.sdk.internal.session.contentscanning.tasks.DefaultScanMediaTask
+import org.matrix.android.sdk.internal.session.contentscanning.tasks.DownloadEncryptedTask
+import org.matrix.android.sdk.internal.session.contentscanning.tasks.GetServerPublicKeyTask
+import org.matrix.android.sdk.internal.session.contentscanning.tasks.ScanEncryptedTask
+import org.matrix.android.sdk.internal.session.contentscanning.tasks.ScanMediaTask
+import java.io.File
+
+@Module
+internal abstract class ContentScannerModule {
+    @Module
+    companion object {
+
+        @JvmStatic
+        @Provides
+        @ContentScannerDatabase
+        @SessionScope
+        fun providesContentScannerRealmConfiguration(realmKeysUtils: RealmKeysUtils,
+                                                     @SessionFilesDirectory directory: File,
+                                                     @UserMd5 userMd5: String): RealmConfiguration {
+            return RealmConfiguration.Builder()
+                    .directory(directory)
+                    .name("matrix-sdk-content-scanning.realm")
```

Comment:
@bmarty do I have db name to "matrix-sdk-content-scanner.realm" ? This can be a problem for existing database, isn't it ?

Reply:
No, please keep the existing name for the reason you mention. I can live with that filename :)